attr/xattr/attribute.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package xattr
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+)
+
+// ValidateableAttribute defines an interface for validating an attribute value.
+type ValidateableAttribute interface {
```

Comment:
Interface is currently located in `xattr` package to remove import cycles present when it is in `validation` package:

```shell
package github.com/hashicorp/terraform-plugin-framework/datasource
        imports github.com/hashicorp/terraform-plugin-framework/datasource/schema
        imports github.com/hashicorp/terraform-plugin-framework/internal/fwschema
        imports github.com/hashicorp/terraform-plugin-framework/resource/schema/defaults
        imports github.com/hashicorp/terraform-plugin-framework/types
        imports github.com/hashicorp/terraform-plugin-framework/types/basetypes
        imports github.com/hashicorp/terraform-plugin-framework/internal/reflect
        imports github.com/hashicorp/terraform-plugin-framework/types/validation
        imports github.com/hashicorp/terraform-plugin-framework/function
        imports github.com/hashicorp/terraform-plugin-framework/internal/reflect: import cycle not allowed
```

Reply:
An unfortunate consequence of the `reflect` package still using `tftypes` 🙁, not much we can realistically do here so I think leaving it in `xattr` makes sense.

Reply:
I agree with leaving it in the `xattr` package.